src/ray/raylet/scheduling_queue.cc

```diff
@@ -223,6 +223,32 @@ void SchedulingQueue::QueueMethodsWaitingForActorCreation(
   QueueTasks(methods_waiting_for_actor_creation_, tasks);
 }
 
+bool SchedulingQueue::TaskInQueue(const TaskID &task_id) const {
+  if (methods_waiting_for_actor_creation_.TaskInQueue(task_id)) {
+    return true;
+  }
+  if (waiting_tasks_.TaskInQueue(task_id)) {
+    return true;
+  }
+  if (placeable_tasks_.TaskInQueue(task_id)) {
+    return true;
+  }
+  if (ready_tasks_.TaskInQueue(task_id)) {
+    return true;
+  }
+  if (running_tasks_.TaskInQueue(task_id)) {
+    return true;
+  }
+  if (blocked_tasks_.TaskInQueue(task_id)) {
+    return true;
+  }
+  return false;
+}
+
+bool SchedulingQueue::TaskQueue::TaskInQueue(const TaskID &task_id) const {
```

Comment:
I think we already have an implementation for this here:
```
bool SchedulingQueue::TaskQueue::HasTask(const TaskID &task_id) const {
  return task_map_.find(task_id) != task_map_.end();
}
```

Reply:
Oops, did not see that. Thanks!